src/decorator-annotator.ts

```diff
@@ -57,7 +57,19 @@ class ClassRewriter extends Rewriter {
     if (decSym.flags & ts.SymbolFlags.Alias) {
       decSym = this.typeChecker.getAliasedSymbol(decSym);
     }
-    return decSym.getDocumentationComment().some(c => c.text.indexOf('@Annotation') >= 0);
+    for (let d of decSym.getDeclarations()) {
+      // yuk, getLeadingCommentRangesOfNode is a private API, using it until
+      // https://github.com/Microsoft/TypeScript/issues/7393 is resolved.
+      // Switch to the TS JSDoc parser in the future to avoid false positives here.
```

Comment:
See getJSDoc() in tsickle.ts for another approach that doesn't use private API.  It's about as long as what you wrote anyway.